hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpUniqueTokensSubcommand.java

```diff
@@ -0,0 +1,286 @@
+        final var uniques = gatherUniques(uniquesStore);
+        System.out.printf("    %d unique tokens gathered%n", uniques.size());
+
+        int reportSize;
+        try (@NonNull final var writer = new Writer(uniquesPath)) {
+            if (emitSummary == EmitSummary.YES) reportSummary(writer, uniques);
+            reportOnUniques(writer, uniques);
+            reportSize = writer.getSize();
+        }
+
+        System.out.printf("=== uniques report is %d bytes%n", reportSize);
+    }
+
+    record UniqueNFTId(long id, long serial) implements Comparable<UniqueNFTId> {
+
+        static UniqueNFTId from(@NonNull final UniqueTokenKey ukey) {
+            return new UniqueNFTId(ukey.getNum(), ukey.getTokenSerial());
+        }
+
+        static UniqueNFTId from(@NonNull final EntityNumPair enp) {
+            return new UniqueNFTId(enp.getHiOrderAsLong(), enp.getLowOrderAsLong());
+        }
+
+        @Override
+        public String toString() {
+            return "%d%s%d".formatted(id, FIELD_SEPARATOR, serial);
+        }
+
+        @Override
+        public int compareTo(UniqueNFTId o) {
+            return ComparisonChain.start()
+                    .compare(this.id, o.id)
+                    .compare(this.serial, o.serial)
+                    .result();
+        }
+    }
+
+    @SuppressWarnings(
+            "java:S6218") // "Equals/hashcode method should be overridden in records containing array fields" - this
+    // record will never be compared or used as a key
+    record UniqueNFT(
+            EntityId owner,
+            EntityId spender,
+            @NonNull RichInstant creationTime,
+            @NonNull byte[] metadata,
+            @NonNull NftNumPair previous,
+            @NonNull NftNumPair next) {
+
+        static final byte[] EMPTY_BYTES = new byte[0];
+
+        static UniqueNFT from(@NonNull final UniqueTokenValue utv) {
+            return new UniqueNFT(
+                    utv.getOwner(),
+                    utv.getSpender(),
+                    utv.getCreationTime(),
+                    null != utv.getMetadata() ? utv.getMetadata() : EMPTY_BYTES,
+                    utv.getPrev(),
+                    utv.getNext());
+        }
+
+        static UniqueNFT from(@NonNull final MerkleUniqueToken mut) {
+            return new UniqueNFT(
+                    mut.getOwner(),
+                    mut.getSpender(),
+                    mut.getCreationTime(),
+                    null != mut.getMetadata() ? mut.getMetadata() : EMPTY_BYTES,
+                    mut.getPrev(),
+                    mut.getNext());
+        }
+    }
+
+    void reportSummary(@NonNull final Writer writer, @NonNull final Map<UniqueNFTId, UniqueNFT> uniques) {
+        final var relatedEntityCounts = RelatedEntities.countRelatedEntities(uniques);
+        writer.writeln("=== %7d unique tokens".formatted(uniques.size()));
+        writer.writeln("    %7d null or missing owners, %7d null or missing spenders"
```

Comment:
Per below (`0.0.0` means treasury-owned), `null or missing owners` might be better put as `held by treasuries` or something like that.

Reply:
See above comment for how it looks now.